### PR TITLE
[luci/pass] Revise ResolveCustomOpAddPass to introduce num_of_broadcast_to

### PR DESCRIPTION
**Pull request by Seunghui98**
This commit revises ResolveCustomOpAddPass to introduce num_of_broadcast_to.

ONE-DCO-1.0-Signed-off-by: SeungHui Lee <shsh1004.lee@samsung.com>

---
Related to Issue: https://github.com/Samsung/ONE/issues/10784
Draft PR: [#10737](https://github.com/Samsung/ONE/pull/11701)

### CODE REVIEW

File: compiler/luci/pass/src/ResolveCustomOpAddPass.cpp [hunk 1]
```diff
@@ -25,30 +25,49 @@
 namespace
 {
 
+/// @brief Returns the number of BroadcastTo(Custom) node among cop's inputs
```
Comment (Seunghui98) [inline compiler/luci/pass/src/ResolveCustomOpAddPass.cpp:28]: This PR comes from the following discussion.
- https://github.com/Samsung/ONE/pull/12509#discussion_r1470611294

File: compiler/luci/pass/src/ResolveCustomOpAddPass.cpp [hunk 2]
```diff
@@ -25,30 +25,49 @@
...
+{
+  int32_t bt_cnt = 0;
+  for (uint32_t idx = 0; idx < cop->numInputs(); idx++)
+  {
+    auto input = loco::must_cast<const luci::CircleNode *>(cop->inputs(idx));
+    switch (input->opcode())
+    {
+      case luci::CircleOpcode::CIRCLECUSTOMOUT:
+      {
+        auto inputOut = loco::must_cast<const luci::CircleCustomOut *>(cop->inputs(idx));
```
Comment (seanshpark) [inline compiler/luci/pass/src/ResolveCustomOpAddPass.cpp:39]:
```suggestion
        auto inputOut = loco::must_cast<const luci::CircleCustomOut *>(input);
```

File: compiler/luci/pass/src/ResolveCustomOpAddPass.cpp [hunk 3]
```diff
@@ -25,30 +25,49 @@
...
+        auto broadcastTo = loco::must_cast<luci::CircleCustom *>(inputOut->input());
+        if (broadcastTo->custom_code() == "BroadcastTo")
```
Comment (seanshpark) [inline compiler/luci/pass/src/ResolveCustomOpAddPass.cpp:41]:
```suggestion
        auto custom = loco::must_cast<luci::CircleCustom *>(inputOut->input());
        if (custom->custom_code() == "BroadcastTo")
```

File: compiler/luci/pass/src/ResolveCustomOpAddPass.cpp (hunk 1 above)
Comment (seanshpark) [inline compiler/luci/pass/src/ResolveCustomOpAddPass.cpp:28]: `BroadcastTo(Custom)` is strange.
- As this is `CircleCustom` Op, Custom("BroadcastTo") would be better

Comment (Seunghui98) [reply compiler/luci/pass/src/ResolveCustomOpAddPass.cpp]: @seanshpark 

Thank you for your comments !
I reflected these in this PR, 
so Could you review the code again? :)

File: compiler/luci/pass/src/ResolveCustomOpAddPass.cpp [hunk 4]
```diff
@@ -25,30 +25,49 @@
 namespace
 {
 
+/// @brief Returns the number of Custom("BroadcastTo") node among cop's inputs
+int32_t num_of_broadcast_to(luci::CircleCustom *cop)
```
Comment (seanshpark) [inline compiler/luci/pass/src/ResolveCustomOpAddPass.cpp:29]:
```suggestion
int32_t num_of_broadcast_to(const luci::CircleNode *node)
```
from the functionality of this method, parameter need not to be `CircleCustom`
and would be better `const` type.

Comment (Seunghui98) [reply compiler/luci/pass/src/ResolveCustomOpAddPass.cpp]: @seanshpark 

Oh, Thank you for letting me know.
I reflected it. :)

File: compiler/luci/pass/src/ResolveCustomOpAddPass.cpp [hunk 5]
```diff
@@ -25,30 +25,49 @@
 namespace
 {
 
+/// @brief Returns the number of Custom("BroadcastTo") node among cop's inputs
```
[the remark this answers is not in the document]
Comment (Seunghui98) [reply compiler/luci/pass/src/ResolveCustomOpAddPass.cpp]: @seanshpark 
Okay. I revised this comment. :)

**Comment by Seunghui98** [comment]
@mhs4670go 

Could you review this PR? :)

File: compiler/luci/pass/src/ResolveCustomOpAddPass.cpp [hunk 6]
```diff
@@ -25,30 +25,49 @@
 namespace
 {
 
+/// @brief Returns the number of Custom("BroadcastTo") node among node's inputs
```
Comment (seanshpark) [inline compiler/luci/pass/src/ResolveCustomOpAddPass.cpp:28]: `node among node's inputs`

> but the word node already exist that may confuse understanding.

Comment (Seunghui98) [reply compiler/luci/pass/src/ResolveCustomOpAddPass.cpp]: @seanshpark 

Oh, I'm sorry I missed the comment in the back. 😅

I uploaded it after modifying it as follows, is it okay?

> /// @brief Returns the number of Custom("BroadcastTo") among node's inputs

**Review by seanshpark — APPROVED** [review]
LGTM

**Comment by Seunghui98** [comment]
@mhs4670go 

Could you review this PR? :)

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**